src/tig.c

```diff
@@ -728,6 +727,12 @@ main(int argc, const char *argv[])
 
 	while (view_driver(display[current_view], request)) {
 		view = display[current_view];
+
+		if (saved_search && *saved_search) {
+			string_copy(view->env->search, saved_search);
```

Comment:
This can be moved out of the loop and copy directly to `argv_env`. To keep things simpler we can probably just do it in the readline version of `prompt_init`.